app/javascript/packs/homePageFeed.jsx

```diff
@@ -0,0 +1,81 @@
+import { h, render } from 'preact';
+import { Article, LoadingArticle } from '../articles';
+import { FeaturedArticle } from '../articles/FeaturedArticle';
+import { Feed } from './Feed.jsx.erb';
+
+/**
+ * Sends analytics about the featured article.
+ *
+ * @param {number} articleId
+ */
+function sendFeaturedArticleAnalytics(articleId) {
+  (function logFeaturedArticleImpression() {
+    if (!window.ga || !ga.create) {
+      setTimeout(logFeaturedArticleImpression, 20);
+      return;
+    }
+
+    ga(
+      'send',
+      'event',
+      'view',
+      'featured-feed-impression',
+      `articles-${articleId}`,
+      null,
+    );
+  })();
+}
+
+const FeedLoading = () => (
+  <div>
+    <LoadingArticle />
+    <LoadingArticle />
+    <LoadingArticle />
+  </div>
+);
+
+/**
+ * Renders the main feed.
+ */
+export const renderFeed = timeFrame => {
+  const feedContainer = document.getElementById('homepage-feed');
+
+  render(
+    <Feed
+      timeFrame={timeFrame}
+      renderFeed={({ feedItems, feedIcons }) => {
+        if (feedItems.length === 0) {
```

Comment:
We're never going to have an empty <Feed /> since there is so much content on the site, so realistically, when there are no articles, it's just the component loading up and waiting for data from the feed endpoint.